src/main/java/com/github/mkopylec/sessioncouchbase/persistent/CouchbaseSession.java

```diff
@@ -60,12 +60,12 @@ public static String globalAttributeName(String attributeName) {
 
     @Override
     public long getCreationTime() {
-        return round((double) globalAttributes.get(CREATION_TIME_ATTRIBUTE));
+        return round(new Double(globalAttributes.get(CREATION_TIME_ATTRIBUTE).toString()));
```

Comment:
Also consider `((Number) globalAttributes.get(CREATION_TIME_ATTRIBUTE)).doubleValue();` for casting the numbers. This should perform better because no `String`s will be created therefore the JVM strings table will not grow.